kernel/src/actions/deletion_vector.rs

```diff
@@ -223,15 +229,111 @@ pub(crate) fn treemap_to_bools(treemap: RoaringTreemap) -> Vec<bool> {
     }
 }
 
+#[derive(PartialEq)]
+#[repr(C)]
+pub enum VectorType {
+    Selection,
+    Deletion,
+}
+
+pub trait BitSet<V> {
+    fn set_bit(&mut self, idx: usize, value: impl Into<V>);
+}
+
+pub trait ContainerIterator<C, V>
+where
+    C: BitSet<V>,
+{
+    fn next_batch(&mut self, mut container: C, batch_size: usize) -> Option<C> {
+        self.next_batch_with(&mut container, batch_size)
+            .map(|_| container)
+    }
+    fn next_batch_with(&mut self, container: &mut C, batch_size: usize) -> Option<()>;
+}
+
+impl<C, V> BitSet<V> for C
+where
+    C: AsMut<[V]>,
+{
+    fn set_bit(&mut self, idx: usize, value: impl Into<V>) {
+        if let Some(i) = self.as_mut().get_mut(idx) {
+            *i = value.into();
+        } else {
+            tracing::info!("Index value in DV: {} is None", idx);
+        }
+    }
+}
+
+#[repr(C)]
+pub struct DeletionVectorBatch {
+    inner_iter: Peekable<IntoIter>,
+    position: usize,
+    total_rows: usize,
+    vector_type: VectorType,
```

Comment:
why not just store `bit_value: bool`, and then you can get rid of `flip_value` all together and do:
```rust
 container.set_bit(bit as usize - self.position, self.bit_value);
```

below